### PR TITLE
Update yii.js

### PR DESCRIPTION
**Pull request by DrDeath72**
add empty url and fix hash

### CODE REVIEW

**Comment by SilverFire** [comment]
Could you elaborate in details, what this PR enhances/fixes, please?

**Comment by DrDeath72** [comment]
1. yii.getQueryParams() use page url by default, not error
2. fix hash error {page: "4#list"}

File: framework/assets/yii.js [hunk 1]
```diff
@@ -272,12 +272,13 @@ yii = (function ($) {
         },
 
         getQueryParams: function (url) {
+            url = url || document.location.href;
```
Comment (klimov-paul) [inline framework/assets/yii.js]: Are we allow calling `getQueryParams` without arguments?
This change will just hide possible developer mistake.
Does not seem to be right.

File: framework/assets/yii.js [hunk 2]
```diff
@@ -272,12 +272,13 @@ yii = (function ($) {
...
             var pos = url.indexOf('?');
             if (pos < 0) {
                 return {};
             }
 
-            var pairs = url.substring(pos + 1).split('&'),
+            var pairs = url.substring(pos + 1).split('#')[0].split('&'),
```
Comment (klimov-paul) [inline framework/assets/yii.js]: Make sense. Still need to test it out.

Comment (SilverFire) [inline framework/assets/yii.js]: Works as expected :+1:

File: framework/assets/yii.js (hunk 1 above)
Comment (SilverFire) [inline framework/assets/yii.js]: Agree on this, should be removed.

**Comment by SilverFire** [comment]
We need a CHANGELOG line

File: framework/assets/yii.js (hunk 1 above)
Comment (DrDeath72) [inline framework/assets/yii.js]: noooooooooooooooo

Comment (SilverFire) [inline framework/assets/yii.js]: [No!](http://www.nooooooooooooooo.com/) :)
Share your thoughts, please. What is your use case?

Comment (DrDeath72) [inline framework/assets/yii.js]: don't write every time document.location.href to work with current url

**Comment by SilverFire** [comment]
Merged https://github.com/yiisoft/yii2/commit/08b9f12afa39ed55c0f35e67502bc23a1d4ff1f2

**Comment by SilverFire** [comment]
Thank you!

**Pull request closed without merging**